service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -424,187 +404,68 @@ public static List<String> validateRequestInputsAndOutputs(
     return errorList;
   }
 
-  private WdsRecordResponseDetails fetchWdsRecords(RunSetRequest request, BearerToken userToken) {
-    String recordType = request.getWdsRecords().getRecordType();
-
-    ArrayList<RecordResponse> recordResponses = new ArrayList<>();
-    HashMap<String, String> recordIdsWithError = new HashMap<>();
-    for (String recordId : request.getWdsRecords().getRecordIds()) {
-      try {
-        recordResponses.add(wdsService.getRecord(recordType, recordId, userToken));
-      } catch (WdsServiceApiException e) {
-        log.warn("Record lookup for Record ID {} failed.", recordId, e);
-        recordIdsWithError.put(recordId, WdsClientUtils.extractErrorMessage(e.getMessage()));
-      } catch (WdsServiceException e) {
-        log.warn("Record lookup for Record ID {} failed.", recordId, e);
-        recordIdsWithError.put(recordId, e.getMessage());
-      }
-    }
-
-    return new WdsRecordResponseDetails(recordResponses, recordIdsWithError);
-  }
-
-  private RunStateResponse storeRun(
-      UUID runId,
-      String externalId,
-      RunSet runSet,
-      String recordId,
-      CbasRunStatus runState,
-      String errors) {
-    String additionalErrorMsg = "";
-    int created =
-        runDao.createRun(
-            new Run(
-                runId,
-                externalId,
-                runSet,
-                recordId,
-                DateUtils.currentTimeInUTC(),
-                runState,
-                DateUtils.currentTimeInUTC(),
-                DateUtils.currentTimeInUTC(),
-                errors));
+  private RunSet registerRunSet(
```

Comment:
TOL this might be a good candidate of business logic to move to the `RunSetService` / `RunSetHandler` layer